pocs/utils/google/storage.py

```diff
@@ -88,3 +91,239 @@ def upload(self, local_path, remote_path=None):
+
+        Args:
+            remote_blob (google.cloud.storage.blob.Blob): Blob object
+                corresponding to stored FITS file.
+            parse_line (callable, optional): A callable function which will be
+                passed an 80 character string corresponding to each line in the header.
+
+        Returns:
+            dict: FITS header as a dictonary.
+        """
+        i = 1
+        if remote_blob.name.endswith('.fz'):
+            i = 2  # We skip the compression header info
+
+        headers = dict()
+
+        streaming = True
+        while streaming:
+            # Get a header card
+            start_byte = 2880 * (i - 1)
+            end_byte = (2880 * i) - 1
+            b_string = remote_blob.download_as_string(start=start_byte, end=end_byte)
+
+            # Loop over 80-char lines
+            for j in range(0, len(b_string), 80):
+                item_string = b_string[j:j + 80].decode()
+
+                # End of FITS Header, stop streaming
+                if item_string.startswith('END'):
+                    streaming = False
+                    break
+
+                # If custom parse function supplied, call that
+                if parse_line is not None:
+                    try:
+                        parse_line(item_string)
+                    except TypeError as e:
+                        warn('parse_line must be callable')
+                    continue
+
+                # Get key=value pairs (skip COMMENTS and HISTORY)
+                if item_string.find('=') > 0:
+                    k, v = item_string.split('=')
+
+                    # Remove FITS comment
+                    if ' / ' in v:
+                        v = v.split(' / ')[0]
+
+                    v = v.strip()
+
+                    # Cleanup and discover type in dumb fashion
+                    if v.startswith("'") and v.endswith("'"):
+                        v = v.replace("'", "").strip()
+                    elif v.find('.') > 0:
+                        v = float(v)
+                    elif v == 'T':
+                        v = True
+                    elif v == 'F':
+                        v = False
+                    else:
+                        v = int(v)
+
+                    headers[k.strip()] = v
+
+            i += 1
+
+        return headers
+
+
+def upload_observation_to_bucket(pan_id,
+                                 dir_name,
+                                 include_files='*.fz',
+                                 bucket='panoptes-survey',
+                                 **kwargs):
+    """Upload an observation directory to google cloud storage.
+
+    This is a convenience function for bulk uploading an observation folder to a
+    bucket. This assumes that observations are placed within `/images/fields`
+    and follow the normal naming convention for observations.
+
+    Note:
+        This requires that the command line utility `gsutil` be installed
```

Comment:
It does seem odd that we are taking two approaches to dealing with GCS, esp. authorization. Should we just require that `gcloud login` (or `gcloud ...service_account...`) has been done?

Reply:
The `gsutil` is part of `gcloud` so I don't think they are mutually exclusive. 

Agree with the authentication as noted above, so we'll move toward the `gcloud` service key authorization.